src/include/com/amazonaws/kinesis/video/common/Include.h

```diff
@@ -482,6 +487,7 @@ struct __RequestInfo {
                                               //!< NOTE: The body will follow the main struct
     UINT32 bodySize;                          //!< Size of the body in bytes
     CHAR url[MAX_URI_CHAR_LEN + 1];           //!< The URL for the request
+    PCHAR service;                            //!< The AWS service for the request

```

Comment:
Strongly recommend having a char array of size MAX_SERVICE_NAME_LEN and copy into it with zero termination and checking for overflow. This is important for multiple reasons, including the encapsulation and deep isolation which can make the structure easy to pass around boundaries and execution units (including different threads)